react-sample-app/src/App.js

```diff
@@ -1,11 +1,62 @@
 import React from 'react';
-import Welcome from './Welcome'
+import logo from './logo.svg';
+import './App.css';
+import AppID from 'ibmcloud-appid-js';
+
+function App() {
+
+  const appID = React.useMemo(() => {
+    return new AppID()
+  }, []);
+
+  const [errorState, setErrorState] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  (async () => {
+    try {
+      await appID.init({
+        clientId: '<SPA_CLIENT_ID>',
+        discoveryEndpoint: '<WELL_KNOWN_ENDPOINT>'
+      });
+    } catch (e) {
+      setErrorState(true);
+      setErrorMessage(e.message);
+    }
+  })();
+
+  const [welcomeDisplayState, setWelcomeDisplayState] = React.useState(false);
+  const [loginButtonDisplayState, setLoginButtonDisplayState] = React.useState(true);
+  const [userName, setUserName] = React.useState('');
+
+  const loginAction = async () => {
+    try {
+      const tokens = await appID.signin();
+      setErrorState(false);
+      setLoginButtonDisplayState(false);
+      setWelcomeDisplayState(true);
+      setUserName(tokens.idTokenPayload.name);
+    } catch (e) {
+      setErrorState(true);
+      setErrorMessage(e.message);
+    }
+  };
 
-export default function App() {
   return (
-      <>
-        <Welcome />
-      </>
+    <div className="App">
+      <header className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        {welcomeDisplayState && <div> Welcome {userName}! You are now authenticated.</div>}
+        {loginButtonDisplayState &&
+        <button
+            style={{fontSize: '24px', backgroundColor: 'skyblue', border: 'none'}}
```

Comment:
see comment above

Reply:
This one has already been published, we discussed that this would be easier for the reader.